src/api/spec/controllers/webui/package_controller_spec.rb

```diff
@@ -822,6 +823,43 @@ def do_request(params)
         it { expect(flash[:error]).not_to be_nil }
         it { expect(response).to redirect_to(package_show_path(source_project, source_package)) }
       end
+
+      context "with a multibuild package" do
+        let(:params) {
+          { project:    source_project,
+            package:    "#{source_package}:multibuild-package",
+            repository: repo_leap_42_2.name,
+            arch:       architecture.name}
+        }
+
+        before do
+          path = "#{CONFIG['source_url']}/build/#{source_project}/_result?view=status" \
+                 "&package=#{source_package}:multibuild-package&arch=i586&repository=#{repo_leap_42_2}"
+          stub_request(:get, path).and_return(body: %(<resultlist state='123'>
+               <result project='#{source_project}' repository='#{repo_leap_42_2}' arch='i586' code="unpublished" state="unpublished">
+                <status package="#{source_package}:multibuild-package" code="succeeded" />
+               </result>
+              </resultlist>))
+        end
+
+        it "should call 'get_status' with appropriate arguments" do
```

Comment:
Is it possible to write this spec without mocking other parts of the controller and the models? I feel like this spec has too many mocks and so would make any refactoring difficult. Ideally the only thing that should be mocked is the backend and any other external API's that we use.